src/js/facility-locator/components/Listbox.jsx

```diff
@@ -0,0 +1,161 @@
+              className={serviceOptionClasses}
+              key={k}
+              value={k}
+              ref={ elem => { this.props.options[i] = elem; }}
+              id={k}
+              onClick={() => this.props.handleFilterSelect(k, 'service')}>
+              <span className="flex-center">
+                <span className="legend spacer"></span>
+                {benefitsServices[k] || k}
+              </span>
+            </li>
+          );
+        })
+        }
+      </ul>
+    );
+  }
+
+  renderSelectOptionWithIcon = (facilityType, index) => {
+    const defaultSelected = !this.props.currentQuery.facilityType && !facilityType;
+    const isSelected = this.isSelectedOption((facilityType), 'facilityType');
+    const isHovered = isSelected || defaultSelected;
+    const facilityOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true,
+      'is-hovered': isHovered
+    });
+    const facilityIconClasses = classNames({
+      legend: true,
+      spacer: !facilityType,
+      [`${kebabCase(facilityType)}-icon`]: facilityType
+    });
+
+    return (
+      <li
+        role="option"
+        tabIndex={this.props.dropdownActive ? '0' : '-1'}
+        aria-selected={isHovered}
+        ref={ elem => { this.props.options[index] = elem; }}
+        id={index > -1 ? (facilityType || 'AllFacilities') : null}
+        className={facilityOptionClasses}
+        onClick={() => this.props.handleFilterSelect(facilityType, 'facility')}
+        onKeyDown={this.props.navigateDropdown}>
+        <span className="flex-center">
+          <span className={facilityIconClasses}></span>
+          {facilityTypes[facilityType] || 'All Facilities'}
+        </span>
+      </li>
+    );
+  }
+
+  renderSelectButtonWithIcon = (facilityType) => {
+    const facilityOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true
+    });
+    const facilityIconClasses = classNames({
+      legend: true,
+      spacer: !facilityType,
+      [`${kebabCase(facilityType)}-icon`]: facilityType
+    });
+
+    return (
+      <div
+        id="facilityDropdown"
+        className={facilityOptionClasses}
+        onKeyDown={this.props.navigateDropdown}>
+        <span className="flex-center">
+          <span className={facilityIconClasses}></span>
+          {facilityTypes[facilityType] || 'All Facilities'}
+        </span>
+      </div>
+    );
+  }
+
+  renderSelectOption = (serviceType, isMobile) => {
+    const serviceOptionClasses = classNames({
+      'dropdown-option': true,
+      'facility-option': true
+    });
+    return (
+      <div className="flex-center">
+        <div
+          id="serviceDropdown"
+          className={serviceOptionClasses}>
+          <span className="flex-center">
+            <span className="legend spacer"></span>
+            {truncate((benefitsServices[serviceType] || serviceType || 'All'), { length: (isMobile ? 38 : 27) })}
```

Comment:
Can we document these magic numbers somehow? With constants and a comment, I think.

Reply:
Good idea.